server/api.proto

```diff
@@ -401,32 +426,53 @@ message UserPresence {
 }
 
 message TTopicJoin {
+  message UserId {
+    repeated bytes user_id = 1;
+  }
+  message Room {
+    repeated bytes room = 1;
+  }
+  message GroupId {
+    repeated bytes group_id = 1;
+  }
+
   oneof id {
-    bytes user_id = 1;
-    bytes room = 2;
-    bytes group_id = 3;
+    UserId user_id = 1;
```

Comment:
Same as adding a friend, how about we have a message with a single `oneof` between user ID, room name, or group ID and repeating that instead. If would allow topic joins for any mix of DM, room, or group topics.